### PR TITLE
Allow YAML input to override config.base and make HPSS_PROJECT configurable

### LINKED ISSUE #1585: Make HPSS_PROJECT configurable like ACCOUNT is
**Description**
Many things are templated in `parm/config/config.base.emc.dyn` but `HPSS_PROJECT` is not. Not all users are able to write to emc-global, this should be configurable like how `ACCOUNT` is.

**Requirements**
Have `HPSS_PROJECT` filled in by `setup_expt.py`

**Acceptance Criteria (Definition of Done)**
When the requirement is met

**(Optional): Suggest A Solution**
- change to a template in `parm/config/config.base.emc.dyn`
- modify `workflow/setup_expt.py` as appropriate to make the default `emc-global` but accept optional values if specified.

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
**Description**

This PR does two main things:

1. Previously, the capability was added to modify placeholder variables like `@DMPDIR@` through an input YAML file when running `setup_expt.py`, but `config.base` was already filled out from the template before this capability was called. This allows for the default values for each host to be overrode by an input YAML file.
2. HPSS_PROJECT was hardcoded in `config.base.emc.dyn` to be `emc-global` and is now configurable, with `emc-global` remaining the default value.

Fixes #1585 

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

- [x] Clone and Build tests on Hera
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: workflow/setup_expt.py [hunk 1]
```diff
@@ -289,6 +291,15 @@ def edit_baseconfig(host, inputs):
...
 
+    # Read in the YAML file to fill out templates and override host defaults
+    yaml_path = inputs.yaml
+    yaml_dict = YAMLFile(path=yaml_path)
+    try:
+        cfg_dict = get_template_dict(yaml_dict['base'])
+    except KeyError:
+        cfg_dict = {}
+    tmpl_dict = dict(tmpl_dict, **cfg_dict)
+
```
Comment (aerorahul) [inline workflow/setup_expt.py]:
```suggestion
    try:
        tmpl_dict = dict(tmpl_dict, **get_template_dict(yaml_dict['base']))
    except KeyError:
        pass
```

Comment (aerorahul) [reply workflow/setup_expt.py]: In line 247, can you accept `yaml_dict`?
```python
def edit_baseconfig(host, inputs, yaml_dict):
```

Comment (aerorahul) [reply workflow/setup_expt.py]: Also, move lines 233, 234 before 231 and update 231 to pass `yaml_dict`?
```python
    yaml_path = inputs.yaml
    yaml_dict = YAMLFile(path=yaml_path)

    edit_baseconfig(host, inputs, yaml_dict)
```

**Review by aerorahul — APPROVED** [review]
looks good.
just a couple of rearrangements so we don't repeat things unnecessarily.

File: workflow/setup_expt.py (hunk 1 above)
Comment (CoryMartin-NOAA) [reply workflow/setup_expt.py]: Ok yeah sure

**Review by aerorahul — APPROVED** [review]
looks good.
thanks for including suggestions.

**Pull request merged**